docs/notebooks/cold_plasma_tensor_elements.ipynb

```diff
@@ -49,248 +58,536 @@
+    }
+   },
+   "outputs": [],
+   "source": [
+    "L_neg, L_pos = filter_negative_and_positive_values(L)\n",
+    "R_neg, R_pos = filter_negative_and_positive_values(R)\n",
+    "P_neg, P_pos = filter_negative_and_positive_values(P)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% raw\n"
+    }
+   },
+   "source": [
+    "Now we can plot the tensor components rotating frame too."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
     }
    },
    "outputs": [],
    "source": [
-    "L_pos = L * (L > 0)\n",
-    "R_pos = R * (R > 0)\n",
-    "L_neg = L * (L < 0)\n",
-    "R_neg = R * (R < 0)\n",
-    "L_pos[L_pos == 0] = np.NaN\n",
-    "R_pos[R_pos == 0] = np.NaN\n",
-    "L_neg[L_neg == 0] = np.NaN\n",
-    "R_neg[R_neg == 0] = np.NaN\n",
-    "\n",
     "plt.figure(figsize=(12, 6))\n",
-    "plt.semilogx(f, abs(L_pos), f, abs(R_pos), f, abs(P_pos), lw=2)\n",
-    "plt.semilogx(\n",
-    "    f,\n",
-    "    abs(L_neg),\n",
-    "    \"#1f77b4\",\n",
-    "    f,\n",
-    "    abs(R_neg),\n",
-    "    \"#ff7f0e\",\n",
-    "    f,\n",
-    "    abs(P_neg),\n",
-    "    \"#2ca02c\",\n",
-    "    lw=2,\n",
-    "    ls=\"--\",\n",
+    "\n",
+    "plt.semilogx(f, abs(L_neg), red, lw=2, ls=\"--\", label=\"L < 0\")\n",
+    "plt.semilogx(f, abs(R_neg), blue, lw=2, ls=\"--\", label=\"R < 0\")\n",
+    "plt.semilogx(f, abs(P_neg), orange, lw=2, ls=\"--\", label=\"P < 0\")\n",
+    "\n",
+    "plt.semilogx(f, L_pos, red, lw=2, label=\"L > 0\")\n",
+    "plt.semilogx(f, R_pos, blue, lw=2, label=\"R > 0\")\n",
+    "plt.semilogx(f, P_pos, orange, lw=2, label=\"P > 0\")\n",
+    "\n",
+    "plt.ylim(ylim)\n",
+    "plt.xlim(f_min, f_max)\n",
+    "\n",
+    "plt.title(\n",
+    "    \"Cold plasma dielectric permittivity tensor components in the rotating frame\",\n",
+    "    size=16,\n",
     ")\n",
+    "\n",
+    "plt.xlabel(\"Frequency [Hz]\", size=16)\n",
+    "plt.ylabel(\"Absolute value\", size=16)\n",
+    "\n",
     "plt.yscale(\"log\")\n",
+    "\n",
     "plt.grid(True, which=\"major\")\n",
     "plt.grid(True, which=\"minor\")\n",
-    "plt.xlim(1e6, 200e9)\n",
-    "plt.legend((\"L > 0\", \"R > 0\", \"P > 0\", \"L < 0\", \"R < 0\", \"P < 0\"), fontsize=16, ncol=2)\n",
-    "plt.xlabel(\"RF Frequency [Hz]\", size=16)\n",
-    "plt.ylabel(\"Absolute value\", size=16)\n",
-    "plt.tick_params(labelsize=14)"
+    "\n",
+    "plt.tick_params(labelsize=14)\n",
+    "\n",
+    "plt.legend(fontsize=18, ncol=2, framealpha=1)\n",
+    "\n",
+    "plt.show()"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "pycharm": {
+     "name": "#%% raw\n"
+    }
+   },
    "source": [
-    "Checks if the values obtained are coherent. They should satisfy\n",
-    "S = (R+L)/2 and D = (R-L)/2\n",
-    "\n"
+    "Let's use the function from earlier to prepare for plotting."
```

Comment:
Delete all remaining cells from this one to the end of the notebook.   They're doing the same calculations and plots as the cells just before them.